### PR TITLE
Shuttle refactor

### LINKED ISSUE #2161: [Newmap] Lighting Overlay Deletion
Something is deleting lighting overlays, and they're failing to GC (and lagging).

Probably related to shuttle/lift movement.

### LINKED ISSUE #2166: [Newmap] Lighting is (partially) broken on shuttles again
ree

### PR DESCRIPTION
**Pull request by Lohikar**
Rewrites the area movement code used by shuttles & elevators in an effort to make it faster, more extensible, and generally easier to read. Also fixes some bugs relating to lighting & moving areas, such as lighting overlays suddenly being teleported into space for absolutely no reason.

Fixes  #2161.
Fixes #2166.

### CODE REVIEW

File: code/controllers/subsystems/machinery.dm [hunk 1]
```diff
@@ -69,6 +70,10 @@
 		if (rcon_update_queued)
 			build_rcon_lists()
 
+		if (powernet_update_queued)
+			makepowernets()
```
Comment (skull132) [inline code/controllers/subsystems/machinery.dm]: Wouldn't it be an idea to swap all refs of `makepowernets` to use this boolean instead? There's roughly 2 more out there in the wild.

Comment (Lohikar) [inline code/controllers/subsystems/machinery.dm]: I might do that in #2175, since that PR deals with power specifically.

Comment (skull132) [inline code/controllers/subsystems/machinery.dm]: There's literally 2 - 3 remaining.

**Comment by Lohikar** [comment]
Leaving the SSatoms one intact since that's removed by the SSpower PR anyways.

**Pull request merged**